Src/FluentAssertions/Collections/GenericCollectionAssertions.cs

```diff
@@ -267,6 +396,46 @@ public AndConstraint<GenericCollectionAssertions<T>> OnlyHaveUniqueItems<TKey>(E
             return new AndConstraint<GenericCollectionAssertions<T>>(this);
         }
 
+        private AndConstraint<GenericCollectionAssertions<T>> NotBeOrderedBy<TSelector>(
+            Expression<Func<T, TSelector>> propertyExpression,
+            IComparer<TSelector> comparer,
+            SortOrder direction,
+            string because,
+            object[] args)
+        {
+            if (comparer == null)
+            {
+                throw new ArgumentNullException(nameof(comparer),
+                    "Cannot assert collection ordering without specifying a comparer.");
+            }
+
+            if (IsValidProperty(propertyExpression, because, args))
+            {
+                ICollection<T> unordered = Subject.ConvertOrCastToCollection();
+
+                if (!unordered.Any())
```

Comment:
❓ How about sequences with a single item?
I guess they are, like empty lists,  also both ordered and unordered at the same time.

Reply:
@jnyrup My initial thought was that such an assertion should `pass` / not throw. 

However, then I think about what problem this poses to the end user - and that would be that their test fails when really their test is invalid, and is either not using the appropriate assertion or does not have the appropriate arrange/act. And in that case, I would prefer the safety net of the failing assertion so I have an opportunity to correct my test.

The above may be confirmation bias because this is also the current behavior 😕

Reply:
Here are some quick tests (if incorporated, please update them to FA style) on how FA currently behaves.
Would this PR break any of those?

```c#
[TestMethod]
public void EmptyCollection_BeInAscendingOrder()
{
    Action act = () => new int[] { }.Should().BeInAscendingOrder();
    act.Should().NotThrow();
}

[TestMethod]
public void EmptyCollection_BeInDescendingOrder()
{
    Action act = () => new int[] { }.Should().BeInDescendingOrder();
    act.Should().NotThrow();
}

[TestMethod]
public void SingleElement_BeInAscendingOrder()
{
    Action act = () => new int[] { 42 }.Should().BeInAscendingOrder();
    act.Should().NotThrow();
}

[TestMethod]
public void SingleElement_BeInDescendingOrder()
{
    Action act = () => new int[] { 42 }.Should().BeInDescendingOrder();
    act.Should().NotThrow();
}

[TestMethod]
public void EmptyCollection_NotBeInAscendingOrder()
{
    Action act = () => new int[] { }.Should().NotBeAscendingInOrder();
    act.Should().Throw<Exception>();
}

[TestMethod]
public void EmptyCollection_NotBeInDescendingOrder()
{
    Action act = () => new int[] { }.Should().NotBeDescendingInOrder();
    act.Should().Throw<Exception>();
}

[TestMethod]
public void SingleElement_NotBeInAscendingOrder()
{
    Action act = () => new int[] { 42 }.Should().NotBeAscendingInOrder();
    act.Should().Throw<Exception>();
}

[TestMethod]
public void SingleElement_NotBeInDescendingOrder()
{
    Action act = () => new int[] { 42 }.Should().NotBeDescendingInOrder();
    act.Should().Throw<Exception>();
}
```

Reply:
@david-a-jetter Have you had a chance to look if the tests above still passes?

Reply:
@jnyrup These tests pass - and I can work to rearrange them so they match in style of the other tests.

However, it seems like a similar test using an anonymous type and specifying the property results in a failure. Looks like more work to do still... different behavior of primitive collections vs. complex types, which is not okay. For example:

```csharp
        [Fact]
        public void When_asserting_single_item_collections_are_not_in_an_descendingly_ordered_collection_are_not_ordered_descending_it_should_succeed()
        {
            //-----------------------------------------------------------------------------------------------------------
            // Arrange
            //-----------------------------------------------------------------------------------------------------------
            var collection = new[]
            {
                new { Text = "b", Numeric = 1 }
            };

            //-----------------------------------------------------------------------------------------------------------
            // Act
            //-----------------------------------------------------------------------------------------------------------
            Action act = () => collection.Should().NotBeInDescendingOrder(o => o.Numeric, Comparer<int>.Default);

            //-----------------------------------------------------------------------------------------------------------
            // Assert
            //-----------------------------------------------------------------------------------------------------------
            act.Should().NotThrow();
        }
```

Reply:
Hmm. If somebody uses these methods, aren't they already expecting a collection? Why else would you use those specific assertions? So if instead of a collection, the test results in an empty collection or a collection with just one item, I would expect it to fail.

Reply:
> Hmm. If somebody uses these methods, aren't they already expecting a collection? Why else would you use those specific assertions? So if instead of a collection, the test results in an empty collection or a collection with just one item, I would expect it to fail.

@jnyrup did you see ⬆️ ?

Reply:
@david-a-jetter I'm not sure what the problem is (or maybe I forgot).

I just tried asserting collection sortedness in NUnit:
empty and single element collections both:
* passes `Assert.That(list, Is.Ordered);`, and
* fails `Assert.That(list, Is.Not.Ordered);`

I came to think of two inherent properties of empty/single element collections.
They are:
* always ordered, and
* both in ascending *and* descending order at the same time.

Reply:
> I came to think of two inherent properties of empty/single element collections.
> They are:
> 
> * always ordered, and
> * both in ascending _and_ descending order at the same time.

I think this is what we should implement.